codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/LegacyEndpointContextSetter.java

```diff
@@ -63,7 +64,13 @@ public void renderPreEndpointResolutionHook(GoSettings settings, GoWriter writer
                 if $T(ctx) {
                     return next.HandleFinalize(ctx, in)
                 }
-                """, SdkGoTypes.Aws.Middleware.GetRequiresLegacyEndpoints);
+                if err := $T(getIdentity(ctx), m.options.AccountIDEndpointMode); err != nil {
```

Comment:
This doesn't really belong in this "legacy context" integration, it should have its own entity in codegen that generates it.